File: 559-maximum-depth-of-n-ary-tree/559-maximum-depth-of-n-ary-tree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root) {
        if (!root)
            return 0;
        int level = 0;
        std::queue<Node *> q;
        q.push(root);
        
        while(q.empty() == false) {
            level++;
            int numNode = q.size();
            for(int i = 0; i < numNode; i++) {
                Node *tmpNode = q.front();
                q.pop();
                for( int j = 0; j < tmpNode->children.size(); j++)
                     q.push(tmpNode->children[j]);
            }  
        }
        return level;
    }
};

File: 559-maximum-depth-of-n-ary-tree/559-maximum-depth-of-n-ary-tree.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Node
        :rtype: int
        """
        if root is None:
            return 0
        
        level = 0;
        q = collections.deque([root])
        
        while len(q) != 0 :
            numNode = len(q)
            level += 1
            for _ in range(numNode):
                node = q.popleft()
                for child in node.children:
                    q.append(child)
        
        return level